streaming/store.go

```diff
@@ -0,0 +1,202 @@
+package streaming
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/spanner"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/common/utils"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/internal"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/profiles"
+)
+
+// Stores the migration job level data post orchestration of a migration job
+type JobExecutionData struct {
+	MigrationJobId         string
+	SpannerDatabaseName    string
+	AggMonitoringResources string
+	IsShardedMigration     bool
+	CreateTimestamp        time.Time
+}
+
+// Stores the shard level execution data post orchestration of a migration job
+type ShardExecutionData struct {
+	MigrationJobId      string
+	DataShardId         string
+	DataflowResources   string
+	DatastreamResources string
+	PubsubResources     string
+	MonitoringResources string
+	CreateTimestamp     time.Time
+}
+
+// PersistJobExecutionData stores all the metadata associated with a job orchestration for a minimal downtime migration in the metadata db. An example of this metadata is job level data such as the spanner database name.
+func PersistJobExecutionData(ctx context.Context, targetProfile profiles.TargetProfile, sourceProfile profiles.SourceProfile, conv *internal.Conv, migrationJobId string, isSharded bool) (err error) {
+	project, instance, dbName, err := targetProfile.GetResourceIds(ctx, time.Now(), sourceProfile.Driver, nil)
+	if err != nil {
+		err = fmt.Errorf("can't get resource ids: %v", err)
+		return err
+	}
+	dbURI := fmt.Sprintf("projects/%s/instances/%s/databases/%s", project, instance, "spannermigrationtool_metadata")
+	client, err := utils.GetClient(ctx, dbURI)
+	if err != nil {
+		err = fmt.Errorf("can't create client for db %s: %v", dbURI, err)
+		return err
+	}
+	defer client.Close()
+	if err != nil {
+		err = fmt.Errorf("can't create database client: %v", err)
+		return err
+	}
+	err = writeJobExecutionMetadata(ctx, migrationJobId, isSharded, dbName, time.Now(), client)
+	if err != nil {
+		err = fmt.Errorf("can't store generated resources for datashard: %v", err)
+		return err
+	}
+	fmt.Printf("Generated resources stored successfully for migration jobId: %s. You can also look at the 'spannermigrationtool_metadata' database in your spanner instance to get this jobId at a later point of time.\n", migrationJobId)
+	return nil
+}
+
+func UpdateJobExecutionDataWithAggregateMonitoringResources(ctx context.Context, targetProfile profiles.TargetProfile, sourceProfile profiles.SourceProfile, conv *internal.Conv, migrationJobId string) error {
```

Comment:
Should this be a more generic function in case in the future we want to update something else as well so this can be done in a single transaction?

Reply:
This is the reason why I made it super specific. We don't know what we will need to update in the future and its hard to define the interface of what all parameters we may need for that update. For example, the new information that needs to go in might not be in `conv`. 

I think we can refactor once we have a usecase but I am also okay with renaming this if you think that is better. Let me know what you think.